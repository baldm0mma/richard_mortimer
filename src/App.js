import React, { Component } from 'react';
import { getAllCharacters } from './apiCalls';
import './App.css';

export class App extends Component {

  componentDidMount = async () => {
    const results = await getAllCharacters();
    console.log(results);
  }

  render = () => {
    return <div>hello world</div>
  }
}

export default App;