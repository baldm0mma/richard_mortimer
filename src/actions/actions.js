export const sendAllCharacters = allCharacters => ({
  type: 'SEND_ALL_CHARACTERS',
  allCharacters
});

export const sendSingleCharacter = character => ({
  type: 'SEND_SINGLE_CHARACTER',
  character
});