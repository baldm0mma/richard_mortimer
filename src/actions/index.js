export const sendAllCharacters = allCharacters => ({
  type: 'SEND_ALL_CHARACTERS',
  allCharacters
});

export const sendSingleCharacter = character => ({
  type: 'SEND_SINGLE_CHARACTER',
  character
});

export const incrementPage = () => ({
  type: 'INCREMENT_PAGE'
});

export const decrementPage = () => ({
  type: 'DECREMENT_PAGE'
});

export const userName = name => ({
  type: 'USER_NAME',
  name
});
