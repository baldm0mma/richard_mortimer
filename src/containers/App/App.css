body {
  background-color: rgb(22, 33, 31);
}

main {
  background-color: rgb(22, 33, 31);
  height: 100vh;
  width: 100%;
}