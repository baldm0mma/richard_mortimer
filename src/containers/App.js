import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAllCharacters } from '../apiCalls/apiCalls';
import { sendAllCharacters } from '../actions';
import { Route } from 'react-router-dom';
import AllCharacters from './AllCharacters';
import './App.css';

export class App extends Component {

  componentDidMount = async () => {
    const results = await getAllCharacters(this.props.page);
    console.log(results);
    this.props.onReceivingAllCharacters(results);
  }

  render = () => {
    return (
      <Route exact path='/main' component={AllCharacters}></Route>
    )
  }
}

export const mapStateToProps = ({ allCharacters, currentCharacter, page }) => ({
  allCharacters,
  currentCharacter,
  page
});

export const mapDispatchToProps = dispatch => ({
  onReceivingAllCharacters: (results) => dispatch(sendAllCharacters(results))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);