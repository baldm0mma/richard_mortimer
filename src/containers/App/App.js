import './App.css';
import { connect } from 'react-redux';
import { Error } from '../404Error/404Error';
import { getAllCharacters } from '../../apiCalls/apiCalls';
import { randomizeNames } from '../../randomNames';
import { Route, Switch } from 'react-router-dom';
import { sendAllCharacters, userName } from '../../actions';
import AllCharacters from '../AllCharacters/AllCharacters';
import Monitor from '../Monitor/Monitor';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import SingleFocus from '../SingleFocus/SingleFocus';

export class App extends Component {
  componentDidMount = async () => {
    const name = await randomizeNames();
    this.props.getUserName(name);
  };

  componentDidUpdate = async () => {
    if (!this.props.allCharacters.length) {
      for (let i = 1; i < 26; i++) {
        const results = await getAllCharacters(i);
        await this.props.onReceivingAllCharacters(results);
      }
    }
  };

  render = () => {
    return (
      <main>
        <Switch>
          <Route exact path='/' component={Monitor} />
          <Route exact path='/criminals' component={AllCharacters} />
          <Route
            path={`/criminals/${this.props.currentCharacter.name}`}
            component={SingleFocus}
          />
          <Route component={Error} />
        </Switch>
      </main>
    );
  };
}

export const mapStateToProps = ({
  userName,
  currentCharacter,
  allCharacters
}) => ({
  userName,
  currentCharacter,
  allCharacters
});

export const mapDispatchToProps = dispatch => ({
  onReceivingAllCharacters: results => dispatch(sendAllCharacters(results)),
  getUserName: name => dispatch(userName(name))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);

App.propTypes = {
  userName: PropTypes.string,
  currentCharacter: PropTypes.object,
  allCharacters: PropTypes.array
};

