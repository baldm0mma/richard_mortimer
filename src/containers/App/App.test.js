import { App, mapStateToProps, mapDispatchToProps } from './App';
import { sendAllCharacters, userName } from '../../actions';
import { shallow } from 'enzyme';
import React from 'react';

describe('App', () => {
  let wrapper;
  describe('Methods and JSX', () => {
    beforeEach(() => {
      const props = {
        userName: 'Steve',
        currentCharacter: {
          name: 'Karl',
          id: 1
        },
        allCharacters: [{ id: 2, name: 'Biff' }, { id: 3, name: 'Marge' }],
        getUserName: jest.fn(),
        onReceivingAllCharacters: jest.fn()
      };
      wrapper = shallow(<App {...props} />);
    });
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should invoke getUserName() on CDM', () => {
      wrapper.instance().componentDidMount();
      expect(wrapper.instance().props.getUserName).toHaveBeenCalled();
    });
    it('should not invoke onReceivingAllCharacters() on CDU if the fetch has already happened', () => {
      wrapper.instance().componentDidUpdate();
      expect(
        wrapper.instance().props.onReceivingAllCharacters
      ).not.toHaveBeenCalled();
    });
    it('should not invoke getAllCharacters() on CDU if the fetch has already happened', () => {
      wrapper.instance().getAllCharacters = jest.fn();
      wrapper.instance().componentDidUpdate();
      expect(
        wrapper.instance().getAllCharacters
      ).not.toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should contain allCharacters, userName, and currentCharacter', () => {
      const mockState = {
        allCharacters: 'all of us!',
        userName: 'Chuck',
        currentCharacter: 'Biff'
      };
      const expected = {
        allCharacters: 'all of us!',
        userName: 'Chuck',
        currentCharacter: 'Biff'
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('mapDispatchToProps.onReceivingAllCharacters should call dispatch', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = sendAllCharacters(undefined);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.onReceivingAllCharacters();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
    it('mapDispatchToProps.getUserName should call dispatch', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = userName(undefined);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getUserName();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
