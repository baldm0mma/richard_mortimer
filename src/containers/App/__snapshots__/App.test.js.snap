// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Methods and JSX should match the snapshot 1`] = `ShallowWrapper {}`;
